src/test/java/plugins/MatrixAuthPluginTest.java

```diff
@@ -48,14 +49,23 @@ public void helloWorld() throws Exception {
         jenkins.login().doLogin("alice");
 
         FreeStyleJob j = jenkins.jobs.create();
+
         j.save();
 
         // if we login as Bob, he shouldn't see the job
         jenkins.login().doLogin("bob");
+
+        //wait for main panel to appear to make sure page is rendered
+        waitFor(by.id("main-panel"));
+
         assertNull(getElement(by.href("job/"+j.name+"/")));
 
         // contorl assertion: alice shoudl see the link
         jenkins.login().doLogin("alice");
+
+        //wait for main panel to appear to make sure page is rendered
+        waitFor(by.id("main-panel"));
```

Comment:
@recena selenium waits for page load by default, but the page having loaded does not guarantee that the browser has finished rendering, especially because of javascript actions that are only started after page load. Therefore it is recommended to wait until an element you know should exist has rendered on the page, before continuing with your test.